src/bindings/js/node/src/infer_request.cpp

```diff
@@ -190,3 +197,82 @@ void InferRequestWrap::infer(const Napi::Object& inputs) {
 Napi::Value InferRequestWrap::get_compiled_model(const Napi::CallbackInfo& info) {
     return CompiledModelWrap::Wrap(info.Env(), _infer_request.get_compiled_model());
 }
+
+struct TsfnContext {
+    TsfnContext(Napi::Env env) : deferred(Napi::Promise::Deferred::New(env)){};
+
+    Napi::Promise::Deferred deferred;
+    ov::InferRequest* _context_ir;
+    std::thread nativeThread;
+
+    Napi::ThreadSafeFunction tsfn;
+    std::vector<ov::Tensor> input_tensors;
+    std::map<std::string, ov::Tensor> result;
+};
+
+void FinalizerCallback(Napi::Env env, void* finalizeData, TsfnContext* context) {
+    context->nativeThread.join();
+    delete context;
+};
+
+void performInferenceThread(TsfnContext* data) {
+    infer_mutex.lock();
+    for (size_t i = 0; i < data->input_tensors.size(); ++i) {
+        data->_context_ir->set_input_tensor(i, data->input_tensors[i]);
+    }
+    data->_context_ir->infer();
+
+    auto compiled_model = data->_context_ir->get_compiled_model().outputs();
+    std::map<std::string, ov::Tensor> outputs;
+
+    for (auto& node : compiled_model) {
+        auto tensor = data->_context_ir->get_tensor(node);
+        auto new_tensor = ov::Tensor(tensor.get_element_type(), tensor.get_shape());
+        tensor.copy_to(new_tensor);
+        outputs.insert({node.get_any_name(), new_tensor});
+    }
+
+    data->result = outputs;
+    infer_mutex.unlock();
+
+    auto callback = [](Napi::Env env, Napi::Function _, TsfnContext* data) {
+        auto m = data->result;
+        auto outputs_obj = Napi::Object::New(env);
+
+        for (const auto& [key, tensor] : m) {
+            auto new_tensor = ov::Tensor(tensor.get_element_type(), tensor.get_shape());
+            tensor.copy_to(new_tensor);
+            outputs_obj.Set(key, TensorWrap::Wrap(env, new_tensor));
+        }
+        data->deferred.Resolve({outputs_obj});
+    };
+
+    data->tsfn.BlockingCall(data, callback);
+    data->tsfn.Release();
+}
+
+
+Napi::Value InferRequestWrap::infer_async(const Napi::CallbackInfo& info) {
+    if (info.Length() != 1) {
+        reportError(info.Env(), "InferAsync method takes as an argument an array or an object.");
+    }
+    Napi::Env env = info.Env();
+
+    auto parsed_input = parse_input_data(info[0]);
+    auto context_data = new TsfnContext(env);
+
+    context_data->input_tensors = parsed_input;
+    context_data->_context_ir = &_infer_request;
+
+    context_data->tsfn = Napi::ThreadSafeFunction::New(env,
+                                                       Napi::Function(),
+                                                       "TSFN",
+                                                       0,
+                                                       1,
+                                                       context_data,
+                                                       FinalizerCallback,
+                                                       (void*)nullptr);
```

Comment:
```suggestion
                                                       nullptr);
```
is cast required? if yes use C++ cast style